### PR TITLE
[Merged by Bors] - Use `memmap2` instead of `memmap`, enable rustls on windows, fix typos in openssl api

### PR DESCRIPTION
**Pull request by ozgrakkurt**

### CODE REVIEW

**Review by sehz — APPROVED** [review]
bump up crate version (patch)

**Comment by ozgrakkurt** [comment]
> bump up crate version (patch)

done

**Comment by morenol** [comment]
Maybe it will fail to compail due to:

https://github.com/infinyon/future-aio/blob/e97957ef12d294ef5dad06e8cfd0812a83bbeb1f/src/rust_tls.rs#L107 ?

**Review by sehz — APPROVED** [review]

**Comment by ozgrakkurt** [comment]
> Maybe it will fail to compail due to:
> 
> https://github.com/infinyon/future-aio/blob/e97957ef12d294ef5dad06e8cfd0812a83bbeb1f/src/rust_tls.rs#L107
> ?

Maybe I fixed it?

**Comment by ozgrakkurt** [comment]
probably should add windows to CI, doing it now

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -13,6 +13,21 @@ concurrency:
   cancel-in-progress: true
 
 jobs:
+  test-windows:
+    name: Run tests on Windows
```
Comment (morenol) [inline .github/workflows/ci.yml:17]: Maybe we could add to the name that this is with rust tls

Comment (ozgrakkurt) [reply .github/workflows/ci.yml:17]: there are no default features so should be fine afaik

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -13,6 +13,21 @@ concurrency:
   cancel-in-progress: true
 
 jobs:
+  test-windows:
```
Comment (morenol) [inline .github/workflows/ci.yml:16]: we should add this to the `needs` section on the `done` job at the end of this file

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -13,6 +13,21 @@ concurrency:
...
+      - name: Install ${{ matrix.rust }}
+        uses: actions-rs/toolchain@v1
+        with:
+          toolchain: ${{ matrix.rust }}
+          profile: minimal
+      - uses: Swatinem/rust-cache@v2
+        with:
+          key: ${{ matrix.os }}-${{ matrix.check }}
+      - name: Test
+        run: cargo test --features rust_tls
```
[the remark this answers is not in the document]
Comment (ozgrakkurt) [reply .github/workflows/ci.yml:30]: there are no default features so should be fine afaik

File: .github/workflows/ci.yml (hunk 1 above)
Comment (ozgrakkurt) [reply .github/workflows/ci.yml]: done

File: .github/workflows/ci.yml (hunk 2 above)
Comment (ozgrakkurt) [reply .github/workflows/ci.yml]: done

**Review by morenol — APPROVED** [review]

**Comment by ozgrakkurt** [comment]
closes https://github.com/infinyon/future-aio/issues/207

**Comment by morenol** [comment]
@sehz could you check that dependency update?

**Review by morenol — APPROVED** [review]
LGTM

File: Cargo.toml [hunk 4]
```diff
@@ -71,7 +71,7 @@ fluvio-async-tls = { version = "0.3.0", optional = true }
 async-std = { version = "1.6.2", default-features = false, optional = true }
 async-io = { version = "1.1.2", optional = true }
 async-net = { version = "1.6.0", optional = true }
-memmap = { version = "0.7.0", optional = true }
+memmap = { package = "memmap2", version = "0.5", optional = true }
```
Comment (morenol) [inline Cargo.toml:74]: I think that maybe we should update this to 
```suggestion
memmap2 = { version = "0.5", optional = true }
```

Comment (ozgrakkurt) [reply Cargo.toml:74]: so we should re-export it as `memmap2` as well?

Comment (ozgrakkurt) [reply Cargo.toml:74]: not sure since it is fork of the same crate

Comment (morenol) [reply Cargo.toml:74]: I think that we should basically update all `memmap` references to `memmap2`. Unless we explicitly expose them in the public api

Comment (morenol) [reply Cargo.toml:74]: So, I think that we should update it in https://github.com/infinyon/future-aio/blob/e97957ef12d294ef5dad06e8cfd0812a83bbeb1f/src/fs/mmap.rs#L13 and also in https://github.com/infinyon/future-aio/blob/e97957ef12d294ef5dad06e8cfd0812a83bbeb1f/Cargo.toml#L49

Comment (ozgrakkurt) [reply Cargo.toml]: yeah makes sense, done

**Review by morenol — APPROVED** [review]

**Comment by ozgrakkurt** [comment]
closes https://github.com/infinyon/future-aio/issues/108

**Comment by ozgrakkurt** [comment]
bors cancel

**Pull request closed without merging**